driver-core/src/main/com/mongodb/internal/operation/ListCollectionsOperation.java

```diff
@@ -209,6 +231,9 @@ private BsonDocument getCommand() {
         }
         if (nameOnly) {
             command.append("nameOnly", BsonBoolean.TRUE);
+            if (authorizedCollections) {
+                command.append("authorizedCollections", BsonBoolean.TRUE);
+            }
```

Comment:
The documentation says: [_"When used without nameOnly: true, this option has no effect."_](https://docs.mongodb.com/manual/reference/command/listCollections/#:~:text=When%20used%20without%20nameOnly%3A%20true%2C%20this%20option%20has%20no%20effect.) That's why we send the option only if `nameOnly` is `true`.

Reply:
If this changes in the future we'll have to make a code change, for that reason I would include it in the command if its set.

You can use the `putIfTrue` helper to ammend the command document.

Reply:
Done.